wrapper.sh

```diff
@@ -1,120 +1,114 @@
+if [ $? -ne 0 ]; then
+	if [ "x$mam_id" == "x" ]; then
+		echo "no mam_id, and no existing session."
 		exit 1
 	fi
 
-	echo No existing session, creating new cookie file using mam_id from environment
-	curl -s -b mam_id=${mam_id} -c ${COOKIEFILE} https://t.myanonamouse.net/json/dynamicSeedbox.php > /tmp/MAM.output
-	grep '"Success":true' /tmp/MAM.output > /dev/null 2>/dev/null
-  	if [ $? -ne 0 ]
-  	then
-		echo mam_id passed on command line is invalid
+	echo "No existing session, creating new cookie file using mam_id from environment"
+	curl -s -b mam_id=${mam_id} -c ${COOKIE_FILE} https://t.myanonamouse.net/json/dynamicSeedbox.php > $RESPONSE_FILE
+
+	grep '"Success":true' $RESPONSE_FILE > /dev/null 2>/dev/null
+  if [ $? -ne 0 ]; then
+		echo "mam_id passed on command line is invalid"
 		exit 1
 	else
-		grep mam_id ${COOKIEFILE} > /dev/null 2>/dev/null
-		if [ $? -ne 0 ]
-		then
-			echo Command successful, but failed to create cookie file.
+		grep mam_id ${COOKIE_FILE} > /dev/null 2>/dev/null
+		if [ $? -ne 0 ]; then
+			echo "Command successful, but failed to create cookie file."
 			exit 1
 		else
-			echo New session created.
+			echo "New session created."
 		fi
 	fi
 else
-	curl -s -b ${COOKIEFILE} -c ${COOKIEFILE} https://t.myanonamouse.net/json/dynamicSeedbox.php > /tmp/MAM.output
-	grep '"Success":true' /tmp/MAM.output > /dev/null 2>/dev/null
-  	if [ $? -ne 0 ]
-  	then
-		echo response: `cat /tmp/MAM.output`
-		echo Current cookie file is invalid.  Please delete it, set the mam_id, and restart the container.
+	curl -s -b $COOKIE_FILE -c $COOKIE_FILE https://t.myanonamouse.net/json/dynamicSeedbox.php > $RESPONSE_FILE
+	grep '"Success":true' $RESPONSE_FILE > /dev/null 2>/dev/null
+  	if [ $? -ne 0 ]; then
+		echo "Response: `cat $RESPONSE_FILE`"
+		echo "Current cookie file is invalid.  Please delete it, set the mam_id, and restart the container."
 		exit 1
 	else
-		echo Session is valid
+		echo "Session is valid"
 	fi
 
 fi
 
-OLDIP=`cat $CACHEFILE 2>/dev/null`
+OLD_IP=`cat $OLD_IP_FILE 2>/dev/null`
 
-while [ $PPID -ne 1 ]
-do
-	OLDIP=`cat $CACHEFILE 2>/dev/null`
+while [ $PPID -ne 1 ]; do
+	OLD_IP=`cat $OLD_IP_FILE 2>/dev/null`
+	NEW_IP=`curl -s ip4.me/api/ | md5sum - | awk '{print $1}'`
 
-	NEWIP=`curl -s ip4.me/api/ | md5sum - | awk '{print $1}'`
-	if [ "x$DEBUG" != "x" ]
-	then
-		echo Current IP:  `curl -s ip4.me/api/`
+	if [ "x$DEBUG" != "x" ]; then
+		echo "Current IP:  `curl -s ip4.me/api/`"
 	fi
 	
 	# Check to see if the IP address has changed
-	if [ "${OLDIP}" != "${NEWIP}" ]
-	then
-  		echo New IP detected
-  		curl -s -b $COOKIEFILE -c $COOKIEFILE https://t.myanonamouse.net/json/dynamicSeedbox.php > /tmp/MAM.output
+	if [ "$OLD_IP" != "$NEW_IP" ]; then
+    echo "New IP detected"
+    # Save cookie jar to temporary file, to not overwrite
+    # with empty cookies on failed requests
+    curl -s -b $COOKIE_FILE -c $TEMP_COOKIE_FILE https://t.myanonamouse.net/json/dynamicSeedbox.php > $RESPONSE_FILE
```

Comment:
This is the crux of the fix. When using the `-c` flag with `curl`, it write all the cookies to the file passed here. Here's a relevant snippet from `man curl`:
```
       -c, --cookie-jar <filename>
              (HTTP) Specify to which file you want curl to write all cookies after a completed operation. curl  writes
              all  cookies  from  its  in-memory  cookie storage to the given file at the end of operations. Even if no
              cookies are known, a file is created so that it removes any formerly existing cookies from the file.  The
              file  uses  the  Netscape cookie file format. If you set the filename to a single minus, "-", the cookies
              are written to stdout.
```